src/peft/tuners/tuners_utils.py

```diff
@@ -32,6 +35,54 @@
 logger = logging.getLogger(__name__)
 
 
+@contextmanager
+def onload_layer(layer):
+    r"""
+    Moves a module's sub-modules to the execution device before context, after remakes the model state dict if
```

Comment:
I think the docstring could be clarified a bit. This is only for modules with offloaded weights, right? So maybe worth mentioning that this context is a no-op for weights without offloading.

"after" => "afterwards" or "after that"?
"remakes" => "restores"?

Reply:
I re-worded the docstring, hopefully that is better now.  Indeed this context manager does nothing if all the module's sub-modules are onloaded, although we call it on all tuner layers in case they have offloaded components.